docs/standard/native-interop/comwrappers-source-generation.md

```diff
@@ -0,0 +1,198 @@
+
+## Implicit HRESULTs and PreserveSig
+
+COM methods in C# have a different signature than the native methods. Standard COM has a return type of `HRESULT`, a 4 byte integer type representing error and success states. This `HRESULT` return value is hidden by default in the C# signature and converted to an exception when an error value is returned. The last "out" parameter of the native COM signature may optionally be converted into the return in the C# signature.
+
+For example, the following snippets show C# method signatures and the corresponding native signature the generator infers.
+
+```csharp
+void Method1(int i);
+
+int Method2(float i);
+```
+
+```c
+HRESULT Method1(int i);
+
+HRESULT Method2(float i, _Out_ int* returnValue);
+```
+
+If you want to handle the `HRESULT` yourself, you can use the <xref:System.Runtime.InteropServices.PreserveSigAttribute> on the method to indicate the generator should not do this transformation. Below are snippets that demonstrate what native signature the generator expects when `[PreserveSig]` is applied. COM methods must return `HRESULT`, so the return value of any method with `PreserveSig` should be `int`.
+
+```csharp
+[PreserveSig]
+int Method1(int i, out int j);
+
+[PreserveSig]
+int Method2(float i);
+```
+
+```c
+HRESULT Method1(int i, int* j);
+
+HRESULT Method2(float i);
+```
+
+For more information, see [Implicit method signature translations in .NET interop](./preserve-sig.md)
+
+## Incompatibilities and differences to built-in COM
+
+### IUnknown only
+
+The only supported interface base is [IUnknown](/windows/win32/api/unknwn/nn-unknwn-iunknown). Interfaces with an <xref:System.Runtime.InteropServices.InterfaceTypeAttribute> with a value other values than <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIUnknown> are not supported in source-generated COM. Any interfaces without an `InterfaceTypeAttribute` are assumed to derive from `IUnknown`. This differs from built-in COM where the default is <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsDual>.
+
+## Marshalling defaults and support
+
+Source-generated COM has some different default marshalling behaviors from built in COM.
```

Comment:
```suggestion
Source generated COM has some different default marshalling behaviors from built-in COM.
```

Reply:
We use source-generated in most other docs and analyzer warnings.

```suggestion
Source-generated COM has some different default marshalling behaviors from built-in COM.
```